feat: add experimentalDisableStreaming to react integration (#1) (#13036)
withastro/astro: .changeset/three-steaks-exist.md

---
"@astrojs/react": minor
---

> Adds experimental support for disabling streaming

Reviewer: ```suggestion
Adds experimental support for disabling streaming

This is useful to support libraries that are not compatible with streaming such as Stitches.js. To disable streaming for all React components in your project, set `experimentalDisableStreaming: true` as a configuration option for `@astrojs/react`:
```
Not a requirement, but for a minor release, it's fine to add a little more about the feature here! While this doesn't make a great *code* comment, you might want to mention a benefit here in the release notes!

Also might be worth a note/reminder that this turns off ALL streaming for ALL components (I'm assuming? This doesn't make it appear configurable per-component) in case that has effects in someone's project.

Reviewer 2: I'd say "some CSS-in-JS libraries" or something like that, rather than mentioning Stitches.js specifically, as it's likely to be a general issue there.

---
"@astrojs/react": minor
---

Adds experimental support for disabling streaming

> This is useful to support libraries that are not compatible with streaming such as Stitches.js. To disable streaming for all React components in your project, set `experimentalDisableStreaming: true` as a configuration option for `@astrojs/react`:

Author: ```suggestion
This is useful to support libraries that are not compatible with streaming such as some CSS-in-JS libraries. To disable streaming for all React components in your project, set `experimentalDisableStreaming: true` as a configuration option for `@astrojs/react`:
```

---

> "@astrojs/react": patch

Reviewer 2: This adds a feature, so would be a minor

---

> feat: add experimentalDisableStreaming to react integration (#1)

Reviewer 2: ```suggestion
Add support for disabling streaming in React integration
```

Author: @florian-lefebvre Where would I place a code snippet in this repo? How would I go about opening a docs PR? Any advice or direction would be most appreciated.

Author: Ah I figured it out! https://github.com/withastro/docs/pull/10761
